docs/admin/cluster-management.md

```diff
@@ -14,47 +14,141 @@ certainly want the docs that go with that version.</h1>
+* GKE [GKE page](https://cloud.google.com/container-engine),
+* Vagrant [Vagrant setup](../../docs/getting-started-guides/vagrant.md),
+* GCE [GCE setup](../../docs/getting-started-guides/gce.md),
+* CoreOS on Azure [CoreOS on Azure setup](../../docs/getting-started-guides/coreos/azure/README.md),
+* CoreOS on AWS [CoreOS on AWS setup](../../docs/getting-started-guides/coreos.md),
+* CoreOS on VMWare [CoreOS on VMWare setup](../../docs/getting-started-guides/coreos.md).
+
+For more comprehensive and detailed description of cluster setup and configuration process see [From Scratch](../../docs/getting-started-guides/scratch.md) guide.
+
+## Tools for interacting with Kubernetes
+
+Currently Kubernetes has two kinds of tools: developer oriented yet useful for operations people `kubectl` that use user specified configuration/contexts and cluster management scripts live in `cluster` and Kubernetes developer ones in the `hack` folder. To make interacting with Kubernetes simpler you should install `kubectl` in your path and ignore `hack` folder.
+
+Tools in `cluster` folder are mainly `bash` scripts which make managing single cluster easier, but they are currently not designed to allow easy management of multiple clusters from a single machine. If it's necessary it is possible to replicate existing e2e test cluster setup that consists mainly of `hack/ginkgo-e2e.sh` script and special test configs for specific cloud providers e.g. `cluster/gce/config-test.sh`. This means that in general scripts from `cluster` directory find a running cluster by themselves, and you don't need to explicitly specify the cluster you want to modify, as those tools assume that there's the cluster running.
+
+Note the `cluster/kube-env.sh` script that defines the `KUBERNETES_PROVIDER` variable which is used by all other scripts. Default value is `gce` so if you want to manage non-GCE cluster you need to assign an identifier of your provider (e.g. `gke`, `aws`, `azure`, `vagrant`, etc.) to `KUBERNETES_PROVIDER` variable e.g. by putting the line
+```bash
+export KUBERNETES_PROVIDER=vagrant
+```
+to your `.bashrc` or `.profile` file.
 
 ## Upgrading a cluster
 
-The `cluster/kube-push.sh` script will do a rudimentary update; it is a 1.0 roadmap item to have a robust live cluster update system.
+GKE handles cluster upgrades for you, so you don't need to worry about it. If you already have a cluster running in some other environment and want to upgrade Kubernetes version running on it, you can use the `cluster/kube-push.sh` script. It works out of the box if your cluster runs on GCE and some other providers listed in `cluster` folder. Note that `kube-push.sh` requires that cloud providers implement necessary interface, thus not all of them are currently supported.
```

Comment:
To clarify, when I say "I would remove this from your PR for now," I mean "I would remove all the changes you made to the Upgrading a cluster section" (and the deletions of the upgrading API version section)